src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -452,7 +476,9 @@ def _contains_external_cte_ref(self, node: TreeNode, root: TreeNode) -> bool:
 
         return False
 
-    def _is_node_pipeline_breaker(self, node: LogicalPlan) -> bool:
+    def _is_node_pipeline_breaker(
```

Comment:
this function is checking pipeline breaker, for those special, case, maybe get a separate function for it, so that we do not mess the function responsibility. Or since this is just a speical check of select statement, you can just direclty do it at function _is_node_valid_to_breakdown at the end

Reply:
It is not only checking if node is a pipeline breaker but also checking if score is valid and if there is an external CTE ref. There is enough overlap that I can repurpose this function for that.

Reply:
In fact, i looked at the code for _is_node_valid_to_breakdown, you might want to introduce a status called VALID_RELAXED_NODE, and introduce a function called _is_relaxed_pipeline_breaker function, and when _is_relaxed_pipeline_breaker and _is_node_pipeline_breaker are both false, set the valid status to InvalidNodesInBreakdownCategory.NON_PIPELINE_BREAKER, if _is_node_pipeline_breaker is true, keep the original valid status, if _is_node_pipeline_breaker is false, and _is_relaxed_pipeline_breaker is true, set the valid status to VALID_RELAXED_NODE, then at the caller you just need to check the valid status